timeless/restaurants/table_shapes/views.py

```diff
@@ -49,23 +51,31 @@ def create():
 
 @bp.route("/edit/<int:id>", methods=("GET", "POST"))
 def edit(id):
-    """ Edit table shape with id
-    @todo #162:30min Implement edit() method of TableShape model and
-     update template. Use TableShapeForm for that, see how in works in
-     create function. Check and update html template if it's needed.
-    """
+    """Edit table shape with id"""
+    table = models.TableShape.query.get(id)
+    if not table:
+        return abort(HTTPStatus.NOT_FOUND)
+
     if request.method == "POST":
-        flash("Edit not yet implemented")
+        form = forms.TableShapeForm(request.form, instance=table)
+        if form.validate():
+            form.save()
```

Comment:
@krzyk could you please extract the validation and save part and put it in TableShapeForm? That would help reducing the complexity of the method (see also codeclimate error)

Reply:
@emilianodellacasa but in such case how would you like to return the result of both validation and save method?

`validate` returns true/false, `save` returns the saved object
Should I introduce exception in case of invalid validation?

Reply:
@emilianodellacasa @krzyk  I suggest to wait when generic views is done and then fix it there. https://github.com/timelesslounge/timelessis/pull/275/files#diff-52975a5c68b82979e3d8732df5533d4cR124